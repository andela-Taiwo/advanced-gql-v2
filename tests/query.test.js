const gql = require('graphql-tag')
const createTestServer = require('./helper')
const FEED = gql`
  {
    feed {
      id
      message
      createdAt
      likes
      views
    }
  }
`
const USER = gql`
  query {
    user {
      email
      id
    }
  }
`
const POST = gql`
  query {
    post(id: 1) {
      id
      message
      createdAt
      likes
      views
    }
  }
`

const USER_SETTINGS = gql`
  {
    userSettings {
      id
      user{
        email
      }
    }

  }
`

const POSTS = gql`
  query {
    posts {
      id
      message
      createdAt
      likes
      views
    }
  }
`


describe('queries', () => {
  test('feed', async () => {
    const {query} = createTestServer({
      user: {id: 1},
      models: {
        Post: {
          findMany: jest.fn(() => [{id: 2, message: 'new Post', createdAt: 12345839, likes: 28, views: 30}])
        }
      }
    })

    const res = await query({query: FEED})
    expect(res).toMatchSnapshot()
  })
  test('posts', async () => {
    const {query} = createTestServer({
      user: {id: 2},
      models: {
        Post: {
          findMany: jest.fn(() => [{id: 2, message: 'new Post', createdAt: 12345839, likes: 28, views: 30}])
        }
      }
    })

    const res = await query({query: POSTS})
    expect(res).toMatchSnapshot()
  })

  test('post', async () => {
    const {query} = createTestServer({
      user: {id: 2},
      models: {
        Post: {
          findOne: jest.fn(() => [ {id: 1, message: 'new Post', createdAt: 12345839, likes: 28, views: 30}])
        }
      }
    })

    const res = await query({query: POST})
    expect(res).toMatchSnapshot()
  })

  test('userSettings', async () => {
    const {query} = createTestServer({
      user: {id: 1},
      models: {
        Settings: {
          findOne: jest.fn(() => [ {id: 1, user: USER}])
        }
      }
    })

    const res = await query({query: USER_SETTINGS})
    expect(res).toMatchSnapshot()
  })
})
