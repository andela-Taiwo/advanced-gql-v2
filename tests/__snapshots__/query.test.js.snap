// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries feed 1`] = `
Object {
  "data": Object {
    "feed": Array [
      Object {
        "createdAt": "Hello World",
        "id": "1",
        "likes": -85,
        "message": "Hello World",
        "views": 79,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`queries post 1`] = `
Object {
  "data": Object {
    "post": Object {
      "createdAt": "Hello World",
      "id": "82030958-2080-49ac-9ff1-68c064d71659",
      "likes": 11,
      "message": "Hello World",
      "views": -25,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`queries posts 1`] = `
Object {
  "data": Object {
    "posts": Array [
      Object {
        "createdAt": "12345839",
        "id": "2",
        "likes": 28,
        "message": "new Post",
        "views": 30,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`queries userSettings 1`] = `
Object {
  "data": Object {
    "userSettings": Object {
      "id": "50adba6a-a187-4b67-9938-74a5833fd89b",
      "user": Object {
        "email": "Hello World",
      },
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
